components/tailscale/providers.tf

```diff
@@ -0,0 +1,53 @@
+locals {
+  aws_account_id = module.account.outputs.full_account_map[var.stage]
+  assume_role    = format("arn:aws:iam::${local.aws_account_id}:role/${var.namespace}-${var.stage}-%s", var.run_in_cicd ? "terraform" : "admin")
+}
+
+variable "run_in_cicd" {
+  type        = bool
+  description = "If run in CI/CD platform, use terraform delegated role, otherwise use admin."
+  default     = true
+}
+
+variable "assume_role_override" {
+  type        = string
+  description = "Overrides the assumed role for the AWS provider. Useful for local testing and imports."
+  default     = null
+}
```

Comment:
All of this logic is pretty heavily dependent on someone already following the Cloudposse/Atmos account management blueprint, which marries us to their architecture. It makes sense to be prescriptive about their architecture to our customers because we are already deep in it, but public consumers of this "component" will likely not be using that same architecture.

cc @Gowiem

Reply:
Mainly this is a trace of Masterpoint using Spacelift. The variable `run_in_cicd` was renamed from `run_in_spacelift`, for example.
The assumption here is that we still would want to separate local roles and roles used in CI/CD platforms. However, with `tf-controller` this all is very questionable, and the final working version could look very different from the pattern we're following now.